### ISSUE TITLE
Documentation missing: JOI validation result passed directly to handler

### ISSUE THREAD

**Issue by ohcibi** [hapijs/hapi#3864]
When a route's payload gets validated with JOI the `request.payload` field contains the result of `JOI.validation` instead of the validated payload. So instead of accessing `request.payload.foo` one must access `request.payload.value.foo`. It should be mentioned in the documentation.

This might as well be a bug. In `lib/validation.js` on line 92 the value is assigned to `request.payload`. Before the move to `await/async` that line looked identical however due to `value` and `error` are passed to the `postValidate` as separate parameters the symbol `value` contained the payload as expected whereas now that symbol refers to the plain return value of `Joi.validate` which is an object containing the two. I just can't find anything about that move in case it was intended.

**Comment by devinivy** [comment]
Hi @ohcibi—this is almost definitely a product of using an old version of joi with hapi.  Try upgrading to at least joi v12.

**Comment by devinivy** [comment]
(The issue is that old versions of joi do not have a promise-like interface to `Joi.validate()`, and hapi assumes a promise-like interface.)

**Comment by ohcibi** [comment]
@devinivy thanks for jumping in but in fact there was a change needed that I can still find no documentation for. I had my validations written like this:

    payload: { Joi.object({.....

I noticed that he recent examples did not wrap their objects in `Joi.object`, removed it and it worked as before. I did use `Joi.object` because it wasn't working otherwise (can't tell if it was documented that way).
